CorpusCallosum/cc_visualization.py

```diff
@@ -0,0 +1,152 @@
+        thickness_path: Path to thickness_values.txt file
+        measurement_points_path: Path to file containing the original vertex indices where thickness was measured
+        output_dir: Directory for output files
+        resolution: Resolution in mm for the mesh
+        smooth_iterations: Number of smoothing iterations to apply to the mesh
+        colormap: Which colormap to use. Options are:
+        - "red_to_blue": Red -> Orange -> Grey -> Light Blue -> Blue
+        - "blue_to_red": Blue -> Light Blue -> Grey -> Orange -> Red
+        - "red_to_yellow": Red -> Yellow -> Light Blue -> Blue
+        - "yellow_to_red": Yellow -> Light Blue -> Blue -> Red
+
+        color_range: Optional tuple of (min, max) to set fixed color range for the colorbar
+        twoD: If True, generate 2D visualization instead of 3D mesh
+    """
+    # Convert paths to Path objects
+    contours_path = Path(contours_path) if contours_path is not None else None
+    thickness_path = Path(thickness_path)
+    measurement_points_path = Path(measurement_points_path)
+    output_dir = Path(output_dir)
+
+    # Load data and create mesh
+    cc_mesh = CC_Mesh(num_slices=1)  # Will be resized when loading data
+
+    if contours_path is not None:
+        cc_mesh.load_contours(str(contours_path))
+    else:
+        cc_contour, anterior_endpoint_idx, posterior_endpoint_idx = load_fsaverage_cc_template()
+        cc_mesh.contours[0] = np.stack(cc_contour).T
+        cc_mesh.start_end_idx[0] = [anterior_endpoint_idx, posterior_endpoint_idx]
+
+    cc_mesh.load_thickness_values(str(thickness_path), str(measurement_points_path))
+    cc_mesh.set_resolution(resolution)
+
+    if twoD:
+        # cc_mesh.smooth_contour(contour_idx=0, window_size=5)
+        cc_mesh.plot_cc_contour_with_levelsets(
+            contour_idx=0, levelpaths=None, title=None, save_path=str(output_dir / "cc_thickness_2d.png"), colorbar=True
+        )
+    else:
+        cc_mesh.fill_thickness_values()
+        # Create and process mesh
+        cc_mesh.create_mesh(smooth=smooth_iterations, closed=False)
+
+        # Generate visualizations
+        cc_mesh.plot_mesh(
+            colormap=colormap,
+            color_range=color_range,
+            thickness_overlay=True,
+            show_contours=False,
+            show_mesh_edges=True,
+            legend=legend,
+        )
+        cc_mesh.plot_mesh(str(output_dir / "cc_mesh.html"), thickness_overlay=True)
+
+        cc_mesh.plot_cc_contour_with_levelsets(
+            contour_idx=len(cc_mesh.contours) // 2, save_path=str(output_dir / "midslice_2d.png")
+        )
+
+        cc_mesh.to_fs_coordinates(vox_size=[resolution, resolution, resolution])
+        cc_mesh.write_vtk(str(output_dir / "cc_mesh.vtk"))
+        cc_mesh.write_fssurf(str(output_dir / "cc_mesh.fssurf"))
+        cc_mesh.write_overlay(str(output_dir / "cc_mesh_overlay.curv"))
+        cc_mesh.snap_cc_picture(str(output_dir / "cc_mesh_snap.png"))
+
```

Comment:
if you finish successfully, you return 0 (the error code)
```suggestion
   return 0
```